flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/table/descriptors/Elasticsearch.java

```diff
@@ -0,0 +1,389 @@
+		this.bulkFlushBackoffType = ElasticsearchValidator.CONNECTOR_BULK_FLUSH_BACKOFF_TYPE_VALUE_CONSTANT;
+		return this;
+	}
+
+	/**
+	 * Configures how to buffer elements before sending them in bulk to the cluster for efficiency.
+	 *
+	 * <p>Sets an exponential backoff type to use when flushing bulk requests.
+	 */
+	public Elasticsearch bulkFlushBackoffExponential() {
+		this.bulkFlushBackoffType = ElasticsearchValidator.CONNECTOR_BULK_FLUSH_BACKOFF_TYPE_VALUE_EXPONENTIAL;
+		return this;
+	}
+
+	/**
+	 * Configures how to buffer elements before sending them in bulk to the cluster for efficiency.
+	 *
+	 * <p>Sets the maximum number of retries for a backoff attempt when flushing bulk requests.
+	 *
+	 * <p>Make sure to enable backoff by selecting a strategy ({@link #bulkFlushBackoffConstant()} or
+	 * {@link #bulkFlushBackoffExponential()}).
+	 *
+	 * @param maxRetries the maximum number of retries.
+	 */
+	public Elasticsearch bulkFlushBackoffMaxRetries(int maxRetries) {
+		this.bulkFlushBackoffMaxRetries = maxRetries;
+		return this;
+	}
+
+	/**
+	 * Configures how to buffer elements before sending them in bulk to the cluster for efficiency.
+	 *
+	 * <p>Sets the amount of delay between each backoff attempt when flushing bulk requests (in milliseconds).
+	 *
+	 * <p>Make sure to enable backoff by selecting a strategy ({@link #bulkFlushBackoffConstant()} or
+	 * {@link #bulkFlushBackoffExponential()}).
+	 *
+	 * @param delay delay between each backoff attempt (in milliseconds).
+	 */
+	public Elasticsearch bulkFlushBackoffDelay(long delay) {
+		this.bulkFlushBackoffDelay = delay;
+		return this;
+	}
+
+	/**
+	 * Sets connection properties to be used during REST communication to Elasticsearch.
+	 *
+	 * <p>Sets the maximum timeout (in milliseconds) in case of multiple retries of the same request.
+	 *
+	 * @param maxRetryTimeout maximum timeout (in milliseconds)
+	 */
+	public Elasticsearch connectionMaxRetryTimeout(int maxRetryTimeout) {
+		this.connectionMaxRetryTimeout = maxRetryTimeout;
+		return this;
+	}
+
+	/**
+	 * Sets connection properties to be used during REST communication to Elasticsearch.
+	 *
+	 * <p>Adds a path prefix to every REST communication.
+	 *
+	 * @param pathPrefix prefix string to be added to every REST communication
+	 */
+	public Elasticsearch connectionPathPrefix(String pathPrefix) {
+		this.connectionPathPrefix = pathPrefix;
+		return this;
+	}
+
+	@Override
+	public void addConnectorProperties(DescriptorProperties properties) {
+		if (version != null) {
+			properties.putString(CONNECTOR_VERSION(), version);
+		}
+
+		final List<List<String>> hostValues = hosts.stream()
+			.map(host -> Arrays.asList(host.hostname, String.valueOf(host.port), host.schema))
+			.collect(Collectors.toList());
+		properties.putIndexedFixedProperties(
+			CONNECTOR_HOSTS,
+			Arrays.asList(CONNECTOR_HOSTS_HOSTNAME, CONNECTOR_HOSTS_PORT, CONNECTOR_HOSTS_SCHEMA),
+			hostValues);
+
+		if (index != null) {
```

Comment:
Could we keep all of those properties from the start in `DescriptorProperties` and in here just merge them into the `properties` parameter? 

This way we would save some boiler plate code for checking against null. What do you think?